src/app/community/_components/AuthorCard.tsx

```diff
@@ -1,28 +1,101 @@
 import classNames from 'classnames/bind';
+import { MouseEvent, useState } from 'react';
+import { useQueryClient, useMutation } from '@tanstack/react-query';
+import { toast } from 'react-toastify';
 
-import { VerticalTripleDotIcon, myProfileImg } from '@/public/index';
-
+import { VerticalTripleDotIcon } from '@/public/index';
 import ProfileImage from '@/components/ProfileImage/ProfileImage';
+import { deletePost } from '@/api/communityAPI';
 
+import { PopOver } from '@/components';
 import styles from './AuthorCard.module.scss';
 
 const cn = classNames.bind(styles);
 
 interface AuthorCardProps {
+  id?: number;
+  isMine?: boolean;
   nickname: string;
   dateText: string;
+  userImage: string | null;
 }
 
-export default function AuthorCard({ nickname, dateText }: AuthorCardProps) {
+export default function AuthorCard({ id, isMine, nickname, dateText, userImage }: AuthorCardProps) {
+  const queryClient = useQueryClient();
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+
+  const { mutate: deletePostMutation } = useMutation({
+    mutationFn: deletePost,
+    onSuccess: () => {
+      toast.success('리뷰를 삭제되었습니다.');
+      queryClient.invalidateQueries({
+        queryKey: ['MyCustomReview'],
+      });
+    },
+    onError: () => {
+      toast.error('리뷰 삭제 중 오류가 발생하였습니다.');
+    },
+  });
+
+  const handleClickPopup = (e: MouseEvent<SVGElement>) => {
+    e.stopPropagation();
+    setIsPopupOpen(!isPopupOpen);
+  };
+
+  const handleClosePopOver = () => {
+    setIsPopupOpen(false);
+  };
+
+  // const handleClickEdit = (e: MouseEvent<HTMLDivElement>) => {
+  //   e.stopPropagation();
+  // };
+
+  const handleClickDelete = (e: MouseEvent<HTMLDivElement>) => {
+    if (id) {
+      deletePostMutation(id);
+    } else {
+      toast.error('해당 게시글이 존재하지 않습니다. 다시 확인해주세요.');
+    }
+    e.stopPropagation();
+  };
```

Comment:
순서를 바꾸면 코드가 조금 더 깔끔해집니다~!!
```suggestion
  const handleClickDelete = (e: MouseEvent<HTMLDivElement>) => {
    e.stopPropagation();
    if (id) {
      deletePostMutation(id);
      return;
    } 
    toast.error('해당 게시글이 존재하지 않습니다. 다시 확인해주세요.');
  };
```